prefect_gcp/cloud_storage.py

```diff
@@ -468,6 +474,94 @@ def example_cloud_storage_copy_blob_flow():
+    """
+    An enumeration class to represent different file formats,
+    compression options for upload_from_dataframe
+
+    Attributes:
+        CSV: Representation for 'csv' file format with no compression
+            and its related content type and suffix.
+
+        CSV_GZIP: Representation for 'csv' file format with 'gzip' compression
+            and its related content type and suffix.
+
+        PARQUET: Representation for 'parquet' file format with no compression
+            and its related content type and suffix.
+
+        PARQUET_SNAPPY: Representation for 'parquet' file format
+            with 'snappy' compression and its related content type and suffix.
+
+        PARQUET_GZIP: Representation for 'parquet' file format
+            with 'gzip' compression and its related content type and suffix.
+    """
+
+    CSV = ("csv", None, "text/csv", ".csv")
+    CSV_GZIP = ("csv", "gzip", "application/x-gzip", ".csv.gz")
+    PARQUET = ("parquet", None, "application/octet-stream", ".parquet")
+    PARQUET_SNAPPY = (
+        "parquet",
+        "snappy",
+        "application/octet-stream",
+        ".parquet.snappy",
+    )
+    PARQUET_GZIP = ("parquet", "gzip", "application/octet-stream", ".parquet.gz")
+
+    @property
+    def format(self) -> str:
+        """The file format of the current instance."""
+        return self.value[0]
+
+    @property
+    def compression(self) -> Union[str, None]:
+        """The compression type of the current instance."""
+        return self.value[1]
+
+    @property
+    def content_type(self) -> str:
+        """The content type of the current instance."""
+        return self.value[2]
+
+    @property
+    def suffix(self) -> str:
+        """The suffix of the file format of the current instance."""
+        return self.value[3]
+
+    @classmethod
+    def fix_extension_with(cls, gcs_blob: str, ext: str):
+        """Fix the extension of a GCS blob.
+
+        Args:
+            gcs_blob: The path to the GCS blob to be modified.
+            ext: The extension to add to the filename.
+
+        Returns:
+            The modified path to the GCS blob with the new extension.
+        """
+        gcs_blob = Path(gcs_blob)
+        folder = gcs_blob.parent
+        filename = Path(gcs_blob.stem).with_suffix(ext)
+        return str(folder.joinpath(filename))
+
+    @classmethod
+    def lookup(cls, serialization_format: str):
+        """
+        Returns the instance of the class that corresponds to the input `fmt`.
+        If no match is found, returns `CSV_GZIP` as a default value.
+
+        Parameters:
+            serialization_format (str): The string representation of the file format.
+
+        Returns:
+            DataFrameSerializationFormat: The instance of the class that corresponds
+             to `serialization_format`.
+        """
+        try:
+            return cls[serialization_format.upper()]
+        except KeyError:
+            return cls.CSV_GZIP
```

Comment:
I still prefer not to have a lookup method, and simply calling `DataFrameSerializationFormat(serialization_format.upper())`

Also, I don't think it should default to CSV_GZIP if users accidentally have a typo like `PARQUET_SNAPY`